source/COMRegistrationFixes/__init__.py

```diff
@@ -11,71 +11,110 @@
 	@param fileName: the path to the .reg file
 	@type fileName: str
 	"""
 	if not os.path.isfile(fileName):
-		raise FileNotFoundError(f"Cannot apply registry patch, {fileName} not found.")
-	regedit=os.path.join(sysWow64 if wow64 else systemRoot,'regedit.exe')
+		raise FileNotFoundError(f"Cannot apply 64-bit registry patch: {fileName} not found.")
+	# On 64-bit systems, SysWOW64 provides 32-bit apps with a virtual directory to reach 64-bit executables.
+	# For hard coded historical compatibility, regedt32.exe is still the name of the 64-bit app.
+	regedit = os.path.join(sysnative, "regedt32.exe")
 	try:
-		subprocess.check_call([regedit,'/s',fileName])
+		subprocess.check_call([regedit, "/s", fileName])
 	except subprocess.CalledProcessError as e:
-		log.error("Error applying registry patch: %s with %s, %s"%(fileName,regedit,e))
+		log.error(f"Error applying 64-bit registry patch from {fileName} with {regedit}: {e}")
 	else:
-		log.debug("Applied registry patch: %s with %s"%(fileName,regedit))
+		log.debug(f"Applied 64-bit registry patch from {fileName}")
 
 
-OLEACC_REG_FILE_PATH = os.path.join(globalVars.appDir, "COMRegistrationFixes", "oleaccProxy.reg")
 def fixCOMRegistrations():
+	"""Registers most common COM proxies, in case they have accidentally been unregistered or overwritten by
+	3rd party software installs/uninstalls.
 	"""
-	Registers most common COM proxies, in case they had accidentally been unregistered or overwritten by 3rd party software installs/uninstalls.
-	"""
-	is64bit=os.environ.get("PROCESSOR_ARCHITEW6432","").endswith('64')
+	is64bit = os.environ.get("PROCESSOR_ARCHITEW6432", "").endswith("64")
 	winVer = winVersion.getWinVer()
 	OSMajorMinor = (winVer.major, winVer.minor)
-	log.debug("Fixing COM registration for Windows %s.%s, %s"%(OSMajorMinor[0],OSMajorMinor[1],"64 bit" if is64bit else "32 bit"))  
+	log.debug(
+		f"Fixing COM registration for Windows {OSMajorMinor[0]}.{OSMajorMinor[1]}, "
+		"{} bit.".format("64" if is64bit else "32")
+	)
 	# Commands taken from NVDA issue #2807 comment https://github.com/nvaccess/nvda/issues/2807#issuecomment-320149243
+	# And also the discussions in #9039 https://github.com/nvaccess/nvda/issues/9039
 	# OLEACC (MSAA) proxies
-	applyRegistryPatch(OLEACC_REG_FILE_PATH)
+	apply32bitRegistryPatch(OLEACC_REG_FILE_PATH)
 	if is64bit:
-		applyRegistryPatch(OLEACC_REG_FILE_PATH, wow64=True)
+		apply64bitRegistryPatch(OLEACC_REG_FILE_PATH)
 	# IDispatch and other common OLE interfaces
-	registerServer(os.path.join(system32,'oleaut32.dll'))
-	registerServer(os.path.join(system32,'actxprxy.dll'))
+	register32bitServer(os.path.join(system32, "oleaut32.dll"))
+	register32bitServer(os.path.join(system32, "actxprxy.dll"))
 	if is64bit:
-		registerServer(os.path.join(sysWow64,'oleaut32.dll'),wow64=True)
-		registerServer(os.path.join(sysWow64,'actxprxy.dll'),wow64=True)
+		register64bitServer(os.path.join(sysnative, "oleaut32.dll"))

```

Comment:
As we are calling the 64 bit version of regsvr32, the dll path we give it must already assume to be 64 bit. Thus, system32 would be the appropriate directory from regsvr32's point of view. In fact, I don't think c:\windows\sysnative is a valid path unless running as a wow64 32 bit process. Same goes for the next line down as well.